test/test_autograd.py

```diff
@@ -2473,6 +2473,38 @@ def test_symeig_no_eigenvectors(self):
         with self.assertRaisesRegex(RuntimeError, 'backward without computing eigenvectors'):
             torch.autograd.backward([w, v], [torch.ones_like(w), torch.ones_like(v)])
 
+    def test_no_grad_copy(self):
+        # create autograd function that saves grad pointer as class static
+        class MyFunc(Function):
+            static_grad_ptr = None
+
+            @staticmethod
+            def forward(ctx, inp1, inp2):
+                return inp1 + inp2
+
+            @staticmethod
+            def backward(ctx, grad):
+                MyFunc.static_grad_ptr = grad.data_ptr()
+                return grad, grad
+
+        a = torch.randn(5, 6, requires_grad=True)
+        b = torch.randn(5, 6, requires_grad=True)
+        # sum should not trigger no copy since it produce non-contiguous grad
+        MyFunc.apply(a, b).sum().backward()
```

Comment:
You're making that assumption based on the particular implementation of grad of sum. It would be much nicer to implement a function that returns a non-contiguous grad itself.

Reply:
Or just do `.backward(my_non_contig_grad)`.

Reply:
@SsnL It occurs to me that passing in grad to backward manually will change its use_count when we decide whether to clone it, so we are not testing contiguous. So I will go with first approach now.